rustworkx/iterators.pyi

```diff
@@ -0,0 +1,144 @@
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# This file contains only type annotations for PyO3 functions and classes
+# For implementation details, see __init__.py and src/iterators.rs
+
+from typing import (
+    Any,
+    Generic,
+    List,
+    Dict,
+    ItemsView,
+    KeysView,
+    ValuesView,
+    Iterator,
+    Mapping,
+    TypeVar,
+    Tuple,
+    overload,
+    final,
+)
+from abc import ABC
+from collections.abc import Sequence
+from typing_extensions import Self
+
+import numpy as np
+
+S = TypeVar("S")
+T_co = TypeVar("T_co", covariant=True)
+
+__all__ = [
+    "NodeIndices",
+    "PathLengthMapping",
+    "PathMapping",
+    "AllPairsPathLengthMapping",
+    "AllPairsPathMapping",
+    "BFSSuccessors",
+    "EdgeIndexMap",
+    "EdgeIndices",
+    "Chains",
+    "EdgeList",
+    "NodeMap",
+    "NodesCountMapping",
+    "Pos2DMapping",
+    "WeightedEdgeList",
+    "CentralityMapping",
+    "BiconnectedComponents",
+    "ProductNodeMap",
+    "MultiplePathMapping",
+    "AllPairsMultiplePathMapping",
+]
+
+class RustworkxCustomVecIter(Generic[T_co], Sequence[T_co], ABC):
+    def __init__(self) -> None: ...
+    def __eq__(self, other: object) -> bool: ...
```

Comment:
I doubt it matters at all since this is an input to a function that's a black box to Python-based type checkers (so no further operations will be performed on `other`), but it seemed marginally odd to me that this is typed as `object` rather than `Any`.

Reply:
I tried to make it more specific but `mypy.stubgen` was complainining it wasn't the right one. I think `object` just avoids us comparing things to primitive types? It is still very broad

Reply:
Python 3's type system is "pure" OO in the sense that everything inherits from `object`, but the difference from `Any` is that `object` enforces the typing semantics that are associated with an object created by (e.g.) `object()` (effectively you can't do anything with it), while `Any` is the type that allows _any_ operation.

Now I think more about it, I think I can understand mypy would say that this (`__eq__`, particularly) has to be `object` not `Any`: you want to have the type system enforce that you narrow the type to the supported things before starting to access attributes on it.  Using `Any` doesn't do that.  So you and mypy are right here, and my comment was wrong.